### PR TITLE
Enhancement/express middlewares

### PR DESCRIPTION
**Pull request by medhat93**

### CODE REVIEW

File: src/controller/rest-ctrl-method.ts [hunk 1]
```diff
@@ -45,7 +44,7 @@ export class RestCtrlMethod {
 
     this.httpMethod = metadata.controller.getMethodHTTPMethod(this.controllerInfo.classInstance, this.methodInfo.name);
     const httpPath = metadata.controller.getMethodHTTPPath(this.controllerInfo.classInstance, this.methodInfo.name);
-    this.httpPath = this.routePrefix ? this.routePrefix + httpPath : httpPath;
+    this.httpPath = this.routePrefix ? path.join(this.routePrefix, httpPath) : httpPath;
```
Comment (AhmedAli7O1) [inline src/controller/rest-ctrl-method.ts]: @medhat93 why using path? this intended for file system paths and its behavior is depending on the OS you're using (Windows, Linux...etc) while the path you want to join is a URL.

File: src/decorators/http-methods.ts [hunk 2]
```diff
@@ -1,6 +1,22 @@
 import { HttpMethod } from '../enums';
 import * as metadata from '../metadata';
 
+/**
+ * Class Decorator to prefix path for all methods' paths.
```
Comment (AhmedAli7O1) [inline src/decorators/http-methods.ts]: replace with:
Class Decorator to prefix all HTTP methods in a Controller with express compatible HTTP path.

File: src/interfaces/server.ts [hunk 3]
```diff
@@ -1,17 +1,47 @@
 import Joi from '@hapi/joi';
 import multer from 'multer';
+import express from 'express';
+import http from 'http';
 import { IHttpErrorsOptions } from '../errors/interfaces';
-import { Sequence } from '../sequence';
+import serveStatic from 'serve-static';
```
Comment (AhmedAli7O1) [inline src/interfaces/server.ts]: whar are the serve-static and body-parser imports for? why not express.static?

Comment (medhat93) [reply src/interfaces/server.ts]: express depends on serve-static in express.static() func & body-parser in express.urlencoded()

**Pull request merged**